bdpy/dl/torch/models.py

```diff
@@ -367,3 +367,447 @@ def forward(self, z: torch.Tensor) -> torch.Tensor:
+                                stride=1, padding=0)  # 512x8x8
+        self.Rrelu8 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv5 = nn.ConvTranspose2d(
+            512, 256, kernel_size=4, stride=2, padding=1, bias=True)  # 256x12x12
+        self.relu_deconv5 = nn.LeakyReLU(0.3, inplace=True)
+        self.conv5_1 = nn.ConvTranspose2d(
+            256, 256, kernel_size=3, stride=1, padding=1, bias=True)
+        self.relu_conv5_1 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv4 = nn.ConvTranspose2d(
+            256, 128, kernel_size=4, stride=2, padding=1, bias=True)
+        self.relu_deconv4 = nn.LeakyReLU(0.3, inplace=True)
+        self.conv4_1 = nn.ConvTranspose2d(
+            128, 128, kernel_size=3, stride=1, padding=1, bias=True)
+        self.relu_conv4_1 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv3 = nn.ConvTranspose2d(
+            128, 128, kernel_size=4, stride=2, padding=1, bias=True)
+        self.relu_deconv3 = nn.LeakyReLU(0.3, inplace=True)
+        self.conv3_1 = nn.ConvTranspose2d(
+            128, 128, kernel_size=3, stride=1, padding=1, bias=True)
+        self.relu_conv3_1 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv2 = nn.ConvTranspose2d(
+            128, 64, kernel_size=4, stride=2, padding=1, bias=True)
+        self.relu_deconv2 = nn.LeakyReLU(0.3, inplace=True)
+        self.conv2_1 = nn.Conv2d(
+            64, 32, kernel_size=3, stride=1, padding=1, bias=True)  # 32x128x128
+        self.relu_conv2_1 = nn.LeakyReLU(0.3, inplace=True)
+        self.deconv1 = nn.ConvTranspose2d(
+            32, 16, kernel_size=4, stride=2, padding=1, bias=True)  # 3x256x256
+        self.relu_deconv1 = nn.LeakyReLU(0.3, inplace=True)
+        self.conv1_1 = nn.Conv2d(
+            16, 3, kernel_size=3, stride=1, padding=1, bias=True)  # 3x256x256
+        self.conv1_1_tanh = nn.Tanh()  # Hyperbolic Tangent (Tanh) function  # 3x256x256
+        self.rconv = nn.Sequential(
+            self.Rconv6,
+            self.Rrelu6,
+            self.Rconv7,
+            self.Rrelu7,
+            self.Rconv8,
+            self.Rrelu8,
+        ).to(self.__device0)
+        self.deconv = nn.Sequential(
+            self.deconv5,
+            self.relu_deconv5,
+            self.conv5_1,
+            self.relu_conv5_1,
+            self.deconv4,
+            self.relu_deconv4,
+            self.conv4_1,
+            self.relu_conv4_1,
+            self.deconv3,
+            self.relu_deconv3,
+            self.conv3_1,
+            self.relu_conv3_1,
+            self.deconv2,
+            self.relu_deconv2,
+            self.conv2_1,
+            self.relu_conv2_1,
+            self.deconv1,
+            self.relu_deconv1,
+            self.conv1_1,
+            self.conv1_1_tanh,
+        ).to(self.__device1)
+
+    def forward(self, z: torch.Tensor) -> torch.Tensor:
+        f = self.rconv(z)
+        g = self.deconv(f)
+        g = g * 255  # conv1_1_tanh_elt layer's scaling
```

Comment:
Why does the hardcoded rescaling appear in the network module? I had the impression that the scope of responsibilities would differ for this type of rescaling. I felt this type of rescaling should be implemented somewhere different.